File: programmers/python/programmers_dfsbfs_1.py
answer = 0

def dfs(idx, n, num, numbers, target):
	if idx >= n:
		if num == target:
			global answer
			answer += 1
		return
	dfs(idx + 1, n, num + numbers[idx], numbers, target)
	dfs(idx + 1, n, num - numbers[idx], numbers, target)
	return

def solution(numbers, target):
	dfs(0, len(numbers), 0, numbers, target)
	return answer

File: programmers/cpp/programmers_dfsbfs_1.cpp
#include <string>
#include <vector>

using namespace std;

int answer = 0;
void dfs(int idx, vector<int> numbers, int res, int target);

int solution(vector<int> numbers, int target) {
	dfs(0, numbers, 0, target);
	return answer;
}

void dfs(int idx, vector<int> numbers, int res, int target)
{
	if (idx == numbers.size())
	{
		if (res == target)
			answer++;
		return;
	}
	dfs(idx + 1, numbers, res + numbers[idx], target);
	dfs(idx + 1, numbers, res - numbers[idx], target);
}